lib/rust/profiling/src/macros.rs

```diff
@@ -110,34 +25,19 @@ macro_rules! start_interval {
     };
 }
 
-/// End measuring the interval of the given profiling level and name.
-#[doc(hidden)]
-#[macro_export]
-macro_rules! end_interval {
-    ($profiling_level:expr, $interval_name:expr) => {
-        $crate::warn_on_error($crate::mark_end_interval($crate::make_metadata!(
-            $profiling_level,
-            $interval_name
-        )))
-    };
-}
-
 /// Measure an interval with the given profiling level and name that starts before the given
 /// expressions and ends after the given expressions.
 #[doc(hidden)]
 #[macro_export]
 macro_rules! measure_interval {
         ($d:tt, $profiling_level:expr, $interval_name:expr, $($body:tt)*) => {
              {
-                 fn start() {
-                     $crate::start_interval!($profiling_level,$interval_name).release();
-                 }
-
-                 fn end() {
-                     $crate::end_interval!($profiling_level,$interval_name);
-                 }
-
-                 $crate::wrap_block! { start end $($body)* }
+                 let guard = $crate::start_interval!($profiling_level,$interval_name);
+                 let ret = {
+                     $($body)*
+                 };
+                 guard.end();
+                 ret
```

Comment:
Why all of these macros are removed? Your macro works COMPLETELY different than the previous implementation – was this new working style discussed with anyone? Your macro wraps the whole body in `{...}` which prevents the syntax of `let something = foo()?;` to work inside of you macro, for example.